### DISCUSSION TITLE
prepend-sudo

### DISCUSSION THREAD

**Discussion by Eriner** [zimfw/zimfw#16] [Ideas]
It was requested that there be a bind (in the `input` module) to prepend sudo to a command. This isn't a problem, but isn't something I would use. Does anyone have a suggestion for what key combination to bind this to (that doesn't clobber emacs binds)?

**Comment by kylescottmcgill** [comment]
While I read that guys comment, personally i think `sudo !!` is perfectly fine (for forgotten sudo commands), I would rather be explicit than implicit with the key binding, writing out sudo makes me think about the command im running more than a key binding would... Maybe its just me, and being stuck in my old ways...

**Comment by Eriner** [comment]
I use `sudo !!` as well, but as I said, I'm not opposed to adding it. If there are a few people that comment here within the week desiring it, I'll add it. Otherwise, I'm going to leave it be. Thanks for your input.

**Comment by brandon099** [comment]
I have come to prefer double esc to prepend sudo -- oh-my-zsh's [sudo plugin](https://github.com/robbyrussell/oh-my-zsh/blob/master/plugins/sudo/sudo.plugin.zsh) does this. Previous to that, I too just used `sudo !!`, and still do on systems that I can't push my dotfiles to. I :+1: having this functionality.

**Comment by Eriner** [comment]
I will never bind it to `<meta><meta>`. I can see that causing problem for non-emacs users.

**Comment by Eriner** [comment]
I've decided against adding this. I feel that `sudo !!` is sufficient. I forces you to look at your command (one press of enter will only perform expansion, not execution).
